lib/galaxy/authnz/custos_authnz.py

```diff
@@ -390,79 +410,44 @@ def _validate_nonce(self, trans, nonce_hash):
             raise Exception("Nonce mismatch. Check that configured redirect_uri matches the URL you are using.")
 
-    def _load_config_for_cilogon(self):
-        # Set cilogon endpoints
-        self.config["authorization_endpoint"] = "https://cilogon.org/authorize"
-        self.config["token_endpoint"] = "https://cilogon.org/oauth2/token"
-        self.config["userinfo_endpoint"] = "https://cilogon.org/oauth2/userinfo"
-
-    def _load_config_for_custos(self):
-        self.config["well_known_oidc_config_uri"] = self._get_well_known_uri_from_url(self.config["provider"])
-        self.config["credential_url"] = f"{self.config['url'].rstrip('/')}/credentials"
-        self._get_custos_credentials()
-        # Set custos endpoints
-        clientIdAndSec = f"{self.config['client_id']}:{self.config['client_secret']}"
-        eps = requests.get(
-            self.config["well_known_oidc_config_uri"],
-            headers={"Authorization": f"Basic {util.unicodify(base64.b64encode(util.smart_str(clientIdAndSec)))}"},
-            verify=False,
-            params={"client_id": self.config["client_id"]},
-            timeout=util.DEFAULT_SOCKET_TIMEOUT,
-        )
-        well_known_oidc_config = eps.json()
-        self._load_well_known_oidc_config(well_known_oidc_config)
-
-    def _load_config_for_keycloak(self):
-        self.config["well_known_oidc_config_uri"] = self._get_well_known_uri_from_url(self.config["provider"])
-        well_known_oidc_config = self._fetch_well_known_oidc_config(self.config["well_known_oidc_config_uri"])
-        self._load_well_known_oidc_config(well_known_oidc_config)
-
-    def _get_custos_credentials(self):
-        clientIdAndSec = f"{self.config['client_id']}:{self.config['client_secret']}"
-        creds = requests.get(
-            self.config["credential_url"],
-            headers={"Authorization": f"Basic {util.unicodify(base64.b64encode(util.smart_str(clientIdAndSec)))}"},
-            verify=False,
-            params={"client_id": self.config["client_id"]},
-            timeout=util.DEFAULT_SOCKET_TIMEOUT,
-        )
-        credentials = creds.json()
-        self.config["iam_client_secret"] = credentials["iam_client_secret"]
-
-    def _get_well_known_uri_from_url(self, provider):
-        # TODO: Look up this URL from a Python library
-        if provider in ["custos", "keycloak"]:
-            base_url = self.config["url"]
-            # Remove potential trailing slash to avoid "//realms"
-            base_url = base_url if base_url[-1] != "/" else base_url[:-1]
-            return f"{base_url}/.well-known/openid-configuration"
-        else:
-            raise Exception(f"Unknown Custos provider name: {provider}")
-
-    def _fetch_well_known_oidc_config(self, well_known_uri):
+    def _load_config(self,headers : dict = None, params : dict = None):
+        self.config.well_known_oidc_config_uri = self._get_well_known_uri_from_url(self.config.provider)
```

Comment:
The sample config file says that the well known url can be overridden, but that does not appear to have been the case for a while. In fact, I'm wondering whether we need the url parameter at all, and it would be better to deprecate the `url` parameter in favour of the `well_known_oidc_config_uri` parameter, with some backward compatible code for converting the `url` to a well-known one.

Reply:
ping @bernt-matthias @ahmedhamidawan